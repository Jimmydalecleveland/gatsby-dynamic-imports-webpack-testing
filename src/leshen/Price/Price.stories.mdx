import { Story, Meta, Preview, Props, Anchor } from '@storybook/addon-docs/blocks'

import Price, { ticketID } from '.'
import Typography from '../Typography'
import ComponentProgress from '../ComponentProgress'

<Meta title='Components|Price' component={Price} decorators={[(story) => <><ComponentProgress ticketID={ticketID} />{story()}</>]} />

# Price

<Preview>
  <Story name="basic">
    <Price amount="45.99" />
  </Story>
</Preview>

## Props
<Props of={Price} />

## With prefix label
<Preview>
  <Story name="with prefix label">
    <Price prefix="Starting at" amount="54.99" />
  </Story>
</Preview>

## With prefix, suffix, and disclaimer
<Preview>
  <Story name="with prefix, suffix, and disclaimer">
    <Price prefix="Starting at" suffix="Limited time offer" amount="67.99">
      <Typography variant="legal">
        A disclaimer is a common child element to pass to the Price component.
      </Typography>
    </Price>
  </Story>
</Preview>

## Large
<Preview>
  <Story name="large">
    <Price prefix="Starting at" amount="54.99" large />
  </Story>
</Preview>