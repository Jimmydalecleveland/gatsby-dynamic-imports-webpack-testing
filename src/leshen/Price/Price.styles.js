import styled from '@emotion/styled'
import { css } from '@emotion/core'
import { rem } from 'polished'

export const Price = styled.div(
  ({ theme, condensed, large }) => css`
    display: inline-grid;
    grid-template-areas: ${condensed
      ? `
        "prefix prefix prefix"
        "currency dollars change"
        "currency dollars duration"
        "suffix suffix suffix"
      `
      : `
        "prefix prefix prefix prefix"
        "currency dollars change duration"
        "suffix suffix suffix suffix"
      `};
    font-size: ${large
      ? rem(theme.fonts.headings.lg.size[1] * 1.5)
      : rem(theme.fonts.headings.lg.size[1])};
  `
)

export const Prefix = styled.span`
  grid-area: prefix;
  font-size: 1rem;
`

export const Currency = styled.span`
  grid-area: currency;
  font-size: 0.35em;
  line-height: 1;
`

export const DollarAmount = styled.span`
  grid-area: dollars;
  line-height: 0.7;
`

export const ChangeAmount = styled.span(
  ({ condensed }) => css`
    grid-area: change;
    font-size: ${condensed ? '0.35em' : '1em'};
    line-height: 0.7;

    &::before {
      display: ${condensed ? 'none' : 'inline-block'};
      content: '.';
    }
  `
)

export const Duration = styled.span(
  ({ condensed }) => css`
    grid-area: duration;
    font-size: ${condensed ? '0.27em' : '0.4em'};
    line-height: 1;
  `
)

export const Suffix = styled.span`
  grid-area: suffix;
  font-size: 1rem;
`
