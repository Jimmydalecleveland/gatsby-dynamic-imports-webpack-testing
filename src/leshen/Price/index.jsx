import React from 'react'
import PropTypes from 'prop-types'

import * as Styled from './Price.styles'

const Price = ({
  large,
  prefix,
  amount,
  duration,
  suffix,
  children,
  condensed,
  ...props
}) => (
  <>
    <Styled.Price
      condensed={condensed}
      large={large}
      aria-label={`${prefix || ''} $${amount} per ${duration}`}
      {...props}
    >
      {prefix && <Styled.Prefix>{prefix}</Styled.Prefix>}
      <Styled.Currency>$</Styled.Currency>
      <Styled.DollarAmount condensed={condensed}>
        {amount.split('.')[0]}
      </Styled.DollarAmount>
      <Styled.ChangeAmount condensed={condensed}>
        {amount.split('.')[1] || '00'}
      </Styled.ChangeAmount>
      <Styled.Duration condensed={condensed}>{duration}</Styled.Duration>
      {suffix && <Styled.Suffix>{suffix}</Styled.Suffix>}
    </Styled.Price>
    {children && children}
  </>
)

Price.defaultProps = {
  large: false,
  condensed: false,
  prefix: null,
  currency: '$',
  duration: '/mo',
  suffix: null,
  children: null,
}

Price.propTypes = {
  /** Used for prominent pricing, e.g. in the Hero. */
  large: PropTypes.bool,
  condensed: PropTypes.bool,
  prefix: PropTypes.string,
  currency: PropTypes.string,
  amount: PropTypes.string.isRequired,
  duration: PropTypes.string,
  suffix: PropTypes.string,
  /** Commonly used for disclaimers. */
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.arrayOf(PropTypes.element),
  ]),
}

export default Price

export const ticketID = 'CMS-172'
