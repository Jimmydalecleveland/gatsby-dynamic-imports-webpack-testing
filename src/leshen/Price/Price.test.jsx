import React from 'react'
import { render } from 'setup-tests'
import Price from '.'

describe('Price', () => {
  test('Price should render', () => {
    const { container } = render(<Price amount="43.99" />)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('Price should render with a label if provided', () => {
    const { container, getByText } = render(
      <Price prefix="Starting at" amount="43.99" />
    )
    expect(container.firstChild).toMatchSnapshot()
    expect(getByText(/starting at/i))
  })
})
