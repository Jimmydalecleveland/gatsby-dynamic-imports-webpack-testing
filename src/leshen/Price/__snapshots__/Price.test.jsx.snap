// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price Price should render 1`] = `
.emotion-8 {
  display: inline-grid;
  grid-template-areas: "prefix prefix prefix prefix" "currency dollars change duration" "suffix suffix suffix suffix";
  font-size: 3.25rem;
}

.emotion-0 {
  grid-area: currency;
  font-size: 0.35em;
  line-height: 1;
}

.emotion-2 {
  grid-area: dollars;
  line-height: 0.7;
}

.emotion-4 {
  grid-area: change;
  font-size: 1em;
  line-height: 0.7;
}

.emotion-4::before {
  display: inline-block;
  content: '.';
}

.emotion-6 {
  grid-area: duration;
  font-size: 0.4em;
  line-height: 1;
}

<div
  aria-label=" $43.99 per /mo"
  class="emotion-8 emotion-9"
>
  <span
    class="emotion-0 emotion-1"
  >
    $
  </span>
  <span
    class="emotion-2 emotion-3"
  >
    43
  </span>
  <span
    class="emotion-4 emotion-5"
  >
    99
  </span>
  <span
    class="emotion-6 emotion-7"
  >
    /mo
  </span>
</div>
`;

exports[`Price Price should render with a label if provided 1`] = `
.emotion-10 {
  display: inline-grid;
  grid-template-areas: "prefix prefix prefix prefix" "currency dollars change duration" "suffix suffix suffix suffix";
  font-size: 3.25rem;
}

.emotion-2 {
  grid-area: currency;
  font-size: 0.35em;
  line-height: 1;
}

.emotion-4 {
  grid-area: dollars;
  line-height: 0.7;
}

.emotion-6 {
  grid-area: change;
  font-size: 1em;
  line-height: 0.7;
}

.emotion-6::before {
  display: inline-block;
  content: '.';
}

.emotion-8 {
  grid-area: duration;
  font-size: 0.4em;
  line-height: 1;
}

.emotion-0 {
  grid-area: prefix;
  font-size: 1rem;
}

<div
  aria-label="Starting at $43.99 per /mo"
  class="emotion-10 emotion-11"
>
  <span
    class="emotion-0 emotion-1"
  >
    Starting at
  </span>
  <span
    class="emotion-2 emotion-3"
  >
    $
  </span>
  <span
    class="emotion-4 emotion-5"
  >
    43
  </span>
  <span
    class="emotion-6 emotion-7"
  >
    99
  </span>
  <span
    class="emotion-8 emotion-9"
  >
    /mo
  </span>
</div>
`;
